app/Http/Controllers/SportsController.php

```diff
@@ -16,17 +17,48 @@ public function create()
         return view('sports.create', compact('sports', 'countries'));
     }
 
-    public function store(Request $request)
+    /**
+     * Sports store method
+     *
+     * @param \Illuminate\Http\Request $request
+     * @return \Illuminate\Http\RedirectResponse
+     */
+    public function store(Request $request): RedirectResponse
     {
-        // Add your code here
+        $values = $request->except('_token');
+
+        foreach ($values as $key => $countryId) {
+            if ($countryId) {
+                $keyExplode = explode('-', $key);
+
+                $place = $keyExplode[0];
+                $sport = Sport::find($keyExplode[1]);
+
+                $sport->countries()->attach($countryId, ['place' => $place]);
+            }
+        }
 
         return redirect()->route('show');
     }
 
     public function show()
     {
-        // Add your code here
+        $countries = Country::with('sports')->get();
+
+        foreach ($countries as $key => $country) {
+            if (count($country->sports)) {
+                $country->setMedalAmount();
+            } else {
+                $countries->forget($key);
+            }
+        }
+
+        $countries = $countries->sortBy([
```

Comment:
@Evolyuta interesting solution but this can be done in a shorter way, will explain in the video